### ISSUE TITLE
HMAC authentication failed

### ISSUE THREAD

**Issue by nazar-pc** [kylemanna/docker-openvpn#268]
I've installed server as in instructions, generated client certificate and imported .ovpn file in Gnome Network Manager GUI (on Ubuntu 17.10).
However, I can't connect to the server, in logs I see the following:
```
openvpn-server_1  | Mon May 22 13:50:24 2017 Authenticate/Decrypt packet error: packet HMAC authentication failed
openvpn-server_1  | Mon May 22 13:50:24 2017 TLS Error: incoming packet authentication failed from [AF_INET]*.*.*.*:55463
```

I'm wondering if some manual tweaks to Cipher and HMAC Authentication settings are needed:
![2017-05-22 16-53-55](https://cloud.githubusercontent.com/assets/928965/26312117/246da190-3ef6-11e7-826e-34278b018cea.png)

**Comment by kylemanna** [comment]
Here's a [screenshot](http://i.imgur.com/XsnbY0w.png) of my `nm-connection-editor` TLS auth tab, compare to your TLS auth tab.  My security tab matches yours.

**Comment by nazar-pc** [comment]
Thanks, now I've got it working.
Default settings were like following:
![2017-05-22 18-51-21](https://cloud.githubusercontent.com/assets/928965/26317336/a58ec276-3f06-11e7-99fd-b7bb50fe26f8.png)

Changing `Key Direction` to `1` fixes this issue.

Is it possible to add this option to the generated .ovpn file? Otherwise it is quite confusing.

**Comment by kylemanna** [comment]
It is in the config.  Network manager seems to ignore it.  If you know of a way to make network manager behave better, let me know.

**Comment by nazar-pc** [comment]
Here is an upstream issue in Network Manager to track progress: https://bugzilla.gnome.org/show_bug.cgi?id=778154

**Comment by kylemanna** [comment]
Awesome, thanks for the link to the upstream issue.

**Comment by w2ak** [comment]
I tried a slight modification in the config file and discovered that my NetworkManager doesn't ignore the key-direction directive if it is put before.
I thought it would be useful to just change this here instead of waiting for NetworkManager to be fixed. This way it works with older versions as well ;)

see #351

**Comment by kylemanna** [comment]
Attempted to fix in #351
